EOPNOTSUPP can be returned if the filesystem does not support xattrs

gregoster opened the pull request:
<!--
Thank you very much for contributing code or documentation to restic! Please
fill out the following questions to make it easier for us to review your
changes.
-->

What does this PR change? What problem does it solve?
-----------------------------------------------------
EOPNOTSUPP is returned on NetBSD on non-FFSv2ea filesystems, and possibly on other OSes with filesystems which do not support xattrs.  This patch ignores EOPNOTSUPP as an error for xattr.


<!--
Describe the changes and their purpose here, as detailed as needed.
-->

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
https://github.com/restic/restic/issues/5341
<!--
Link issues and relevant forum posts here.

If this PR resolves an issue on GitHub, use "Closes #1234" so that the issue
is closed automatically when this PR is merged.
-->

Checklist
---------

<!--
You do not need to check all the boxes below all at once. Feel free to take
your time and add more commits. If you're done and ready for review, please
check the last box. Enable a checkbox by replacing [ ] with [x].

Please always follow these steps:
- Read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- Enable [maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- Run `gofmt` on the code in all commits.
- Format all commit messages in the same style as [the other commits in the repository](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
-->

- [ ] I have added tests for all code changes.
- [ ] I have added documentation for relevant changes (in the manual).
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I'm done! This pull request is ready for review.

Commit:
EOPNOTSUPP can be returned if the filesystem does not support xattrs

greatroar commented on internal/fs/node_xattr.go:
```diff
@@ -53,9 +53,10 @@ func handleXattrErr(err error) error {
 	case *xattr.Error:
 		// On Linux, xattr calls on files in an SMB/CIFS mount can return
-		// ENOATTR instead of ENOTSUP.
+		// ENOATTR instead of ENOTSUP.  Ignore "Operation not supported"
+		// as well.
```
ENOTSUP is "Operation not supported"; EOPNOTSUPP is "Operation not supported on socket". So I think this comment should call out NetBSD for being weird:

```suggestion
		// ENOATTR instead of ENOTSUP.  NetBSD can return EOPNOTSUPP.
```

MichaelEischer replied:
From the manpage xD . Although, I agree that NetBSD shows some rather unusual behavior here.

> (ENOTSUP and EOPNOTSUPP have the same value on Linux, but according to POSIX.1 these error values should be distinct.)

gregoster replied:
The xattr package for go:
https://github.com/pkg/xattr/blob/master/xattr_test.go
uses both ENOTSUP and EOPNOTSUPP as well.   FreeBSD (I think) will be the same as NetBSD in terms of returning EOPNOTSUPP.  Not sure about Darwin.

MichaelEischer replied:
EOPNOTSUPP seems indeed to be a BSD thing: https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/acl_set.3.html . Although, restic has supported xattrs on darwin, freebsd, linux and solaris for quite a while, and there haven't been any reports regarding that error code. But feel free to replace `NetBSD` with `BSD` in the above suggestion.

MichaelEischer reviewed (changes requested):
See comment above

MichaelEischer commented:
@gregoster are you still interested in completing the PR?

Commit:
Update comment to reflect the BSD family returns EOPNOTSUPP.

Commit:
Add a changelog entry for 5344 - EOPNOTSUPP for xattrs.

MichaelEischer reviewed (approved):
LGTM.

I've rebased the PR to get rid of the merge commit.

MichaelEischer committed:
fs: fix xattr compliation on Linux

On Linux, ENOTSUP and EOPNOTSUPP have the same value such that Go does
not allow adding them to the same case.
